src/librustc_resolve/resolve_imports.rs

```diff
@@ -332,6 +332,7 @@ impl<'a> ::ModuleS<'a> {
 
     fn define_in_glob_importers(&self, name: Name, ns: Namespace, binding: &'a NameBinding<'a>) {
         if !binding.defined_with(DefModifiers::PUBLIC | DefModifiers::IMPORTABLE) { return }
+        let _resolutions = self.resolutions.borrow();
```

Comment:
A comment here, preferably one citing some example source, would be fantastic. e.g. I can kind of guess why this makes sense, but it's not at all obvious.